src/components/inputs.ts

```diff
@@ -38,56 +39,64 @@ export class Input implements Serializable {
 	}
 
 	serialize(): ApplicationCommandOption {
-		return {
+		const payload: ApplicationCommandOption = {
 			type: this.type,
 			name: this.name,
 			description: this.description,
 			required: this.required,
-			// TODO: Implement this
-			// choices?: ApplicationCommandOptionChoice[];
-			// options?: ApplicationCommandOption[];
 		};
+
+		if (this.choices != null) {
+			payload.choices = this.choices.map((choice) => ({
+				name: choice.name,
+				value: choice.value,
+			}));
+		}
+
+		// TODO: Handle Options (do we want to?)
+
+		return payload;
 	}
 }
 
 export class StringInput extends Input {
-	constructor(args: Omit<InputArgs, "type">) {
+	constructor(args: Omit<InputArgs, "type" | "options">) {
 		super({ type: ApplicationCommandOptionType.STRING, ...args });
 	}
 }
 
 export class IntegerInput extends Input {
-	constructor(args: Omit<InputArgs, "type">) {
+	constructor(args: Omit<InputArgs, "type" | "options">) {
 		super({ type: ApplicationCommandOptionType.INTEGER, ...args });
 	}
 }
 
 export class BooleanInput extends Input {
-	constructor(args: Omit<InputArgs, "type">) {
+	constructor(args: Omit<InputArgs, "type" | "choices" | "options">) {
 		super({ type: ApplicationCommandOptionType.BOOLEAN, ...args });
 	}
 }
 
 export class UserInput extends Input {
-	constructor(args: Omit<InputArgs, "type">) {
+	constructor(args: Omit<InputArgs, "type" | "choices" | "options">) {
 		super({ type: ApplicationCommandOptionType.USER, ...args });
 	}
 }
 
 export class ChannelInput extends Input {
-	constructor(args: Omit<InputArgs, "type">) {
+	constructor(args: Omit<InputArgs, "type" | "choices" | "options">) {
 		super({ type: ApplicationCommandOptionType.CHANNEL, ...args });
 	}
 }
 
 export class RoleInput extends Input {
-	constructor(args: Omit<InputArgs, "type">) {
+	constructor(args: Omit<InputArgs, "type" | "choices" | "options">) {
 		super({ type: ApplicationCommandOptionType.ROLE, ...args });
 	}
 }
 
 export class MentionableInput extends Input {
-	constructor(args: Omit<InputArgs, "type">) {
+	constructor(args: Omit<InputArgs, "type" | "choices" | "options">) {
```

Comment:
Would it be better to pull this union unto an alias above?

Reply:
Opted to not do this since it isn't _that_ much repetition, and this just felt a little cleaner to read to me after I had made the change locally. That might just be me being unfamiliar with types though